### PR TITLE
[cryptolib/aes] Driver and test updates

### PR DESCRIPTION
**Pull request by moidx**
* Add static assertions to enum to ensure they are in sync with register definitions. Removed CipherMode switch/case 
statement to be more consistent with production code.
* Switch test case to use a NIST test vector to make the test case easier to audit. The test was previously not passing on Verilator/FPGA targets.

### CODE REVIEW

File: sw/device/lib/crypto/drivers/aes_test.c [hunk 1]
```diff
@@ -9,23 +9,44 @@
 #include "sw/device/lib/testing/test_framework/check.h"
 #include "sw/device/lib/testing/test_framework/ottf_main.h"
 
-static const char kSecretKey[] = "Secret test key.";
+// NIST 800-38a F.5.1 CTR-AES128.Encrypt test vectors.
+static const uint32_t kSecretKey[] = {
```
Comment (moidx) [inline sw/device/lib/crypto/drivers/aes_test.c:13]: I wonder if I should convert these into KATs.

Comment (jadephilipoom) [reply sw/device/lib/crypto/drivers/aes_test.c:13]: We'll eventually need to run the larger ACVP test sets for AES, but I think that should be done through the top-level API so we ensure everything is correct at that entrypoint.

File: sw/device/lib/crypto/drivers/aes_test.c [hunk 2]
```diff
@@ -9,23 +9,44 @@
...
+    {.data = 0xe2bec16b, 0x969f402e, 0x117e3de9, 0x2a179373},
+    // Block#2: ae2d8a571e03ac9c9eb76fac45af8e51
+    {.data = 0x578a2dae, 0x9cac031e, 0xac6fb79e, 0x518eaf45},
+    // Block#3: 30c81c46a35ce411e5fbc1191a0a52ef
+    {.data = 0x461cc830, 0x11e45ca3, 0x19c1fbe5, 0xef520a1a},
+    // Block#4: f69f2445df4f9b17ad2b417be66c3710
+    {.data = 0x45249ff6, 0x179b4fdf, 0x7b412bad, 0x10376ce6},
+};
 
 // AES-CTR encrypted using all zeroes IV.
```
Comment (jadephilipoom) [inline sw/device/lib/crypto/drivers/aes_test.c:40]: IV is no longer all zeroes.
```suggestion
```

File: sw/device/lib/crypto/drivers/aes_test.c (hunk 1 above)
Comment (moidx) [reply sw/device/lib/crypto/drivers/aes_test.c:13]: SG

File: sw/device/lib/crypto/drivers/aes_test.c (hunk 2 above)
Comment (moidx) [reply sw/device/lib/crypto/drivers/aes_test.c]: Done.

**Pull request merged**